### PR TITLE
#166790002 Users Should Be Able to Comment on Articles

### PR DESCRIPTION
**Pull request by elemanhillary-zz**
#### What does this PR do?
- Adds a feature to comment on an article.

#### Description of Task to be completed
- Enable user comment(s) on articles, both authored by the user and those not authored by the user

#### How should this be manually tested?
 ```shell
git clone https://github.com/andela/ah-92explorers-backend.git
cd ah-92explorers-backend
git checkout ft-comment-on-article-166790002
npm install
npm start
```
- Navigate to `localhost:3000/api/articles/:slug/comments`

#### Any background context you want to provide?
- None
#### What are the relevant pivotal tracker stories?
[166790002](https://www.pivotaltracker.com/story/show/166790002)

#### Screenshots (if appropriate)
Imgur

#### Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I don't have more than 3 major commits on this PR

### CODE REVIEW

**Review by IsaiahRn — DISMISSED** [review]
Kindly, make these changes:

1. Change the footer of your commit message, We don't use Maintains on features `[Maintains ##166790002]`, it should be Starts, Finishes or Delivers and keep only one `#`

2. Capitalize every first letter on the PR title

File: controllers/article.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    }, {
+      transaction,
+    });
+    await transaction.commit();
+    if (newArticle) {
+      res.status(201).json({ message: 'article created', article: newArticle, });
+    }
+  } catch (ex) {
+    await transaction.rollback();
+    res.status(400).json({ message: 'article creation failed' });
```
Comment (kodek-sleuth) [inline controllers/article.js]: please use an error key to send an error message

**Review by kodek-sleuth — DISMISSED** [review]

**Review by kodek-sleuth — DISMISSED** [review]

**Review by IsaiahRn — DISMISSED** [review]
This PR is ready to be merged!

**Review by Hervera — DISMISSED** [review]
Ready to be marged

**Review by niyoceles — DISMISSED** [review]

**Review by niyoceles — DISMISSED** [review]
ready to merge

**Review by IsaiahRn — DISMISSED** [review]

**Review by kodek-sleuth — DISMISSED** [review]

File: controllers/comment.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+          message: 'commented',
+          comment: {
+            body: showComment.body,
+            author,
+            createdAt,
+            updatedAt,
+          },
+        });
+      }
+      return;
```
Comment (dmithamo) [inline controllers/comment.js]: Remove unnecessary `return`

File: controllers/comment.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        });
+        const {
+          author, createdAt, updatedAt,
+        } = showComment;
+        return res.status(201).json({
+          message: 'commented',
+          comment: {
+            body: showComment.body,
+            author,
+            createdAt,
```
Comment (dmithamo) [inline controllers/comment.js:44]: Add a reference to the article. i.e,
```
article: {
   slug: `article-slug-2`
}
```

**Review by dmithamo — CHANGES_REQUESTED** [review]

**Review by dmithamo — APPROVED** [review]

**Review by niyoceles — APPROVED** [review]

**Review by niyoceles — APPROVED** [review]

**Review by kodek-sleuth — APPROVED** [review]

**Pull request merged**